shared/src/main/scala/com/raquo/domtypes/generic/defs/eventProps/FormEventProps.scala

```diff
@@ -3,76 +3,83 @@ package com.raquo.domtypes.generic.defs.eventProps
 import com.raquo.domtypes.generic.builders.EventPropBuilder
 
 /**
+  * @tparam DomHtmlElementTargetEvent
+  *            An event that has an HTMLElement as `target`.
+  *            This event type has no corresponding type in JS DOM. See our own `ElementTargetEvent` trait.
+  * @tparam DomFormElementTargetEvent
+  *            An event that has an HTMLFormElement as `target`.
+  *            This event type has no corresponding type in JS DOM. See our own `ElementTargetEvent` trait.
   * @tparam DomInputElementTargetEvent
   *            An event that has an HTMLInputElement as `target`.
-  *            This event type has no corresponding type in JS DOM. See our own `InputElementTargetEvent` trait.
-  *            Strictly speaking, this type is not applicable to `onInput` because the latter
-  *            can be fired on any HTMLElement that has `contentEditable` mode enabled.
+  *            This event type has no corresponding type in JS DOM. See our own `ElementTargetEvent` trait.
+  *            - Strictly speaking, this type is not applicable to `onInput` because the latter
+  *              can be fired on any HTMLElement that has `contentEditable` mode enabled.
+  *            - Similarly for `onChange` and `onSelect` – these could also fire on an `HTMLTextAreaElement`.
   * @tparam DomInputEvent
   *            DOM InputEvent https://developer.mozilla.org/en-US/docs/Web/API/InputEvent
   *            Note: This type is not implemented in scala-js-dom
   */
-trait FormEventProps[EP[_ <: DomEvent], DomEvent, DomFocusEvent <: DomEvent, DomInputEvent <: DomEvent, DomElementTargetEvent <: DomEvent] { this: EventPropBuilder[EP, DomEvent] =>
+trait FormEventProps[EP[_ <: DomEvent], DomEvent, DomFocusEvent <: DomEvent, DomInputEvent <: DomEvent, DomHtmlElementTargetEvent <: DomEvent, DomFormElementTargetEvent <: DomEvent, DomInputElementTargetEvent <: DomEvent] { this: EventPropBuilder[EP, DomEvent] =>
```

Comment:
Maybe we could remove the type bounds from the `Dom...ElementTargetEvent`. When we use it we could always do `...Event with Dom...ElementTargetEvent`. Might be better suited for non-jsdom implementation?

Reply:
Hmm I think for now those type constraints are conceptually sound, both in JS and on the JVM. Event handlers should emit events, this is really all that the constraint says. We can revisit this if/when we get a legit use case where this constraint doesn't make sense.

Practically speaking, if we want to use some trait which is unrelated to `DomEvent`, we could just pass `dom.Event with SomethingElse` as a type param to `FormEventProps` instead of having a `with` inside the latter.